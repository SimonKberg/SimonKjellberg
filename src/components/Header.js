// @flow strict
import * as React from 'react'
import Link from 'next/link'
import styled from '@emotion/styled'

const Wrapper = styled('header')`
  color: #fff;
  background: #000;
  margin-bottom: 1.45rem;
`

const Container = styled('div')`
  margin: 0 auto;
  max-width: 35rem;
  padding: 1.45rem 1.0875rem;
  padding-top: max(1.45rem, env(safe-area-inset-top));
  padding-left: max(1.0875rem, env(safe-area-inset-left));
  padding-right: max(1.0875rem, env(safe-area-inset-right));
`

const Title = styled('h1')`
  font-size: 1rem;
  font-weight: bold;
  margin: 0;
  text-transform: lowercase;
`

const StyledLink = styled('a')`
  color: #fff;
  text-decoration: none;
`

type Props = {
  siteTitle: string,
}

const Header = ({ siteTitle }: Props) => (
  <Wrapper>
    <Container>
      <Title>
        <Link href="/" passHref>
          <StyledLink>
            #!/
            {siteTitle}
          </StyledLink>
        </Link>
      </Title>
    </Container>
  </Wrapper>
)

export default Header
