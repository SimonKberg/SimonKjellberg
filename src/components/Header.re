open Utils;

module Styles = {
  open Css;

  let wrapper =
    style([
      color(Theme.Header.color),
      backgroundColor(Theme.Header.backgroundColor),
      marginBottom(rem(1.45)),
    ]);

  let container =
    style([
      margin2(~v=`zero, ~h=`auto),
      maxWidth(rem(35.0)),
      padding2(~v=rem(1.45), ~h=rem(1.0875)),
      unsafe(
        "@supports(padding: max(0px)) and (padding: env(safe-area-inset-top))",
        style([
          unsafe("paddingTop", "max(env(safe-area-inset-top), 1.45rem)"),
          unsafe("paddingLeft", "max(env(safe-area-inset-left), 1.0875rem)"),
          unsafe(
            "paddingRight",
            "max(env(safe-area-inset-right), 1.0875rem)",
          ),
        ]),
      ),
    ]);

  let title =
    style([
      fontSize(rem(1.0)),
      fontWeight(`bold),
      margin(`zero),
      textTransform(`lowercase),
    ]);

  let link = style([color(Theme.Header.color), textDecoration(`none)]);
};

[@react.component]
let make = (~siteTitle) => {
  <header className=Styles.wrapper>
    <div className=Styles.container>
      <h1 className=Styles.title>
        <Next.Link href="/" passHref=true>
          <a className=Styles.link> {"#!/" ++ siteTitle |> str} </a>
        </Next.Link>
      </h1>
    </div>
  </header>;
};

let default = make;
