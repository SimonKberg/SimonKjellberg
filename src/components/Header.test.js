// @flow strict
import * as React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'

test('<Header />', () => {
  const tree = renderer.create(<Header siteTitle="Site Title" />).toJSON()
  expect(tree).toMatchSnapshot()
})
