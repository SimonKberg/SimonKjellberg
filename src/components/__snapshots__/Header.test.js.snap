// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> 1`] = `
.emotion-3 {
  color: #fff;
  background-color: #000;
  margin-bottom: 1.45rem;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 35rem;
  padding: 1.45rem 1.0875rem;
}

@supports (padding:max(0px)) and (padding:env(safe-area-inset-top)) {
  .emotion-2 {
    padding-top: max(env(safe-area-inset-top),1.45rem);
    padding-left: max(env(safe-area-inset-left),1.0875rem);
    padding-right: max(env(safe-area-inset-right),1.0875rem);
  }
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 700;
  margin: 0px;
  text-transform: lowercase;
}

.emotion-0 {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <h1
      className="emotion-1"
    >
      <a
        className="emotion-0"
        href="/"
        onClick={[Function]}
      >
        #!/Site Title
      </a>
    </h1>
  </div>
</header>
`;
